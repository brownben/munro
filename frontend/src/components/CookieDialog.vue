<!--
  Cookie Law Notice

  Displayed fixed at the top of the page until it is dismissed. When dismissed it slides up and stores the acceptance in local storage.
  Is only displayed if the acceptance is not in local storage.
-->

<template>
  <transition name="slide">
    <div v-if="show" @click="allowCookies()">
      <p>This site uses Cookies and Local Storage in order to function correctly. By using this site you consent to the use of Cookies and Local Storage.</p>
      <svg viewBox="0 0 24 24">
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </div>
  </transition>
</template>

<script>
export default {
  name: 'CookieDialog',

  data: () => ({ show: Boolean }),

  created: function () {
    this.show = (localStorage.getItem('cookies') !== 'accepted')
  },

  methods: {
    allowCookies: function () {
      this.show = false
      localStorage.setItem('cookies', 'accepted')
    },
  },
}
</script>

<style lang="stylus" scoped>
@import '../assets/styles/helpers.styl'

div
  position: fixed
  top: 0
  left: 0
  z-index: 10
  width: 100%
  background-color: purple-400
  color: white
  no-user-select()

  p
    padding: 0.5rem
    width: calc(100% - 2.75rem)
    color: white
    text-align: center

  svg
    position: absolute
    top: 0
    right: 0
    padding: 0 0.5rem
    width: 1.25rem
    height: 100%
    fill: white

.slide-enter-active, .slide-leave-active
  transition: 1s

.slide-enter, .slide-leave-to
  top: -10rem
</style>
