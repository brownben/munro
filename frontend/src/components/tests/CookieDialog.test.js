/*
  Cookie Dialog Component Unit Tests
*/

import { mount } from '@vue/test-utils'
import CookieDialog from '@/components/CookieDialog'

beforeEach(() => localStorage.clear())

test('Is a Vue Instance', () => {
  const wrapper = mount(CookieDialog)
  expect(wrapper.isVueInstance()).toBeTruthy()
})

test('Renders Correctly', () => {
  const wrapper = mount(CookieDialog)
  expect(wrapper.element).toMatchSnapshot()
  wrapper.find('div').trigger('click')
  expect(wrapper.element).toMatchSnapshot()
})

test('Shows when LocalStorage is Empty', () => {
  const wrapper = mount(CookieDialog)
  expect(wrapper.find('p').isVisible()).toBeTruthy()
})

test("Doesn't show when LocalStorage is has Correct Value", () => {
  localStorage.setItem('cookies', 'accepted')
  expect(localStorage.getItem('cookies')).toBe('accepted')
  const wrapper = mount(CookieDialog)
  expect(wrapper.isEmpty()).toBeTruthy()
})

test('Shows when LocalStorage is has Incorrect Value', () => {
  localStorage.setItem('cookies', 'ccepted')
  const wrapper = mount(CookieDialog)
  expect(wrapper.find('p').isVisible()).toBeTruthy()
})

test('Shows Correct Message', () => {
  const wrapper = mount(CookieDialog)
  expect(wrapper.contains('p')).toBeTruthy()
  expect(wrapper.find('p').text()).toBe(
    `This site uses Cookies and Local Storage in order to function
        correctly. By using this site you consent to the use of Cookies and
        Local Storage.`
  )
})

test('Closes When Clicked', () => {
  const wrapper = mount(CookieDialog)
  expect(wrapper.findAll('.card-color').length).not.toBe(0)
  wrapper.find('.card-color').trigger('click')
  expect(wrapper.findAll('.card-color').length).toBe(0)
})
