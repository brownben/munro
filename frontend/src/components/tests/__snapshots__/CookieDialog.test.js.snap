// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<div
  name="slide"
>
  <p>
    This site uses Cookies and Local Storage in order to function correctly. By using this site you consent to the use of Cookies and Local Storage.
  </p>
   
  <svg
    viewBox="0 0 24 24"
  >
    <path
      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
    />
  </svg>
</div>
`;
